mono/metadata/w32process-win32.c

```diff
@@ -230,6 +236,7 @@ process_complete_path (const gunichar2 *appname, gchar **completed)
 
 	*completed = process_quote_path (found);
 exit:
+	MONO_EXIT_GC_SAFE;
 	g_free (found);
```

Comment:
We should be careful to make sure that this `exit` label cannot be reached by some code outside the `MONO_ENTER_GC_SAFE` - or is it safe to call `MONO_EXIT_GC_SAFE` twice?

Reply:
You can't, really - the macro, when paired with a corresponding MONO_ENTER_GC_SAFE, actually expands into a while loop. With that said, at least in general I believe trying to transition to a state you're already in will throw an error.

Reply:
`g_free` can surely lock, though it'll tend not to. It'll tend to just be an interlocked push onto a free list.

Reply:
Agree. the scope on this one feels too big.

Reply:
You mean, because each of the underlying calls might change to not need it, and the need could fall away completely? That would argue for pushing into the callee, except for the layering `g_` problem?